app/models/profile.rb

```diff
@@ -93,6 +93,7 @@ def activate(reason_deactivated: nil)
 
     now = Time.zone.now
     is_reproof = Profile.find_by(user_id: user_id, active: true)
+    is_biometric_upgrade = is_reproof && biometric? && !is_reproof.biometric?
```

Comment:
`is_reproof` sounds like a boolean name, so `is_reproof.biometric?` is unexpected to me? so I'd maybe introduce another variable here to clarify?
```suggestion
    previous_active_profile = Profile.find_by(user_id: user_id, active: true)
    is_reproof = previous_active_profile.present?
    is_biometric_upgrade = is_reproof && biometric? && !previous_active_profile.biometric?
```
could toy around with it more maybe